### ISSUE TITLE
curl url malformed

### ISSUE THREAD

**Issue by gfahy** [ceph/ceph-container#1086]
**Bug Report**

Pulled the ceph/daemon:latest container

What you expected to happen:

That the container would remain running

ubuntu@cephst1:~/ceph_demo$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                 CREATED             STATUS                     PORTS               NAMES
21c888b042d8        ceph/daemon         "/entrypoint.sh demo"   11 minutes ago      Exited (3) 7 minutes ago                       demo
ubuntu@cephst1:~/ceph_demo$

ubuntu@cephst1:~/ceph_demo$ docker logs 21c888b042d8
creating /etc/ceph/ceph.client.admin.keyring
creating /etc/ceph/ceph.mon.keyring
creating /var/lib/ceph/bootstrap-osd/ceph.keyring
creating /var/lib/ceph/bootstrap-mds/ceph.keyring
creating /var/lib/ceph/bootstrap-rgw/ceph.keyring
creating /var/lib/ceph/bootstrap-rbd/ceph.keyring
monmaptool: monmap file /etc/ceph/monmap-ceph
monmaptool: set fsid to fb02d732-2986-4fe0-962b-dad9a5ed0ced
monmaptool: writing epoch 0 to /etc/ceph/monmap-ceph (1 monitors)
importing contents of /var/lib/ceph/bootstrap-osd/ceph.keyring into /etc/ceph/ceph.mon.keyring
importing contents of /var/lib/ceph/bootstrap-mds/ceph.keyring into /etc/ceph/ceph.mon.keyring
importing contents of /var/lib/ceph/bootstrap-rgw/ceph.keyring into /etc/ceph/ceph.mon.keyring
importing contents of /var/lib/ceph/bootstrap-rbd/ceph.keyring into /etc/ceph/ceph.mon.keyring
importing contents of /etc/ceph/ceph.client.admin.keyring into /etc/ceph/ceph.mon.keyring
changed ownership of '/etc/ceph/ceph.client.admin.keyring' from root:root to ceph:ceph
changed ownership of '/etc/ceph/ceph.conf' from root:root to ceph:ceph
ownership of '/etc/ceph/ceph.mon.keyring' retained as ceph:ceph
changed ownership of '/var/lib/ceph/mgr/ceph-cephst1/keyring' from root:root to ceph:ceph
ownership of '/var/lib/ceph/mgr/ceph-cephst1' retained as ceph:ceph
WARNING: it looks like you might be trying to set a ceph-mgr module configuration key.  Since Ceph 13.0.0 (Mimic), mgr module configuration is done with `config set`, and new values set using `config-key set` will be ignored.
set mgr/dashboard/server_addr
changed ownership of '/var/lib/ceph/osd/ceph-0' from root:root to ceph:ceph
/usr/lib/python2.7/site-packages/ceph_disk/main.py:5689: UserWarning: 
*******************************************************************************
This tool is now deprecated in favor of ceph-volume.
It is recommended to use ceph-volume for OSD deployments. For details see:

    http://docs.ceph.com/docs/master/ceph-volume/#migrating

*******************************************************************************

  warnings.warn(DEPRECATION_WARNING)
command: Running command: /usr/bin/ceph-osd --no-mon-config --cluster=ceph --show-config-value=fsid
command: Running command: /usr/bin/ceph-conf --cluster=ceph --name=osd. --lookup bluestore_block_size
command: Running command: /usr/bin/ceph-conf --cluster=ceph --name=osd. --lookup bluestore_block_db_size
command: Running command: /usr/bin/ceph-conf --cluster=ceph --name=osd. --lookup bluestore_block_size
command: Running command: /usr/bin/ceph-conf --cluster=ceph --name=osd. --lookup bluestore_block_wal_size
populate_data_path: Preparing osd data dir /var/lib/ceph/osd/ceph-0
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/ceph_fsid
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/ceph_fsid
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/fsid
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/fsid
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/magic
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/magic
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/type
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/type
/usr/lib/python2.7/site-packages/ceph_disk/main.py:5721: UserWarning: 
*******************************************************************************
This tool is now deprecated in favor of ceph-volume.
It is recommended to use ceph-volume for OSD deployments. For details see:

    http://docs.ceph.com/docs/master/ceph-volume/#migrating

*******************************************************************************

  warnings.warn(DEPRECATION_WARNING)
ownership of '/var/lib/ceph/osd/ceph-0/ceph_fsid' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/magic' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/fsid' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/type' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0' retained as ceph:ceph
/usr/lib/python2.7/site-packages/ceph_disk/main.py:5689: UserWarning: 
*******************************************************************************
This tool is now deprecated in favor of ceph-volume.
It is recommended to use ceph-volume for OSD deployments. For details see:

    http://docs.ceph.com/docs/master/ceph-volume/#migrating

*******************************************************************************

  warnings.warn(DEPRECATION_WARNING)
main_activate: path = /var/lib/ceph/osd/ceph-0
activate: Cluster uuid is fb02d732-2986-4fe0-962b-dad9a5ed0ced
command: Running command: /usr/bin/ceph-osd --no-mon-config --cluster=ceph --show-config-value=fsid
activate: Cluster name is ceph
activate: OSD uuid is 9fdce86f-5b14-453d-b78c-cee90efca8b4
allocate_osd_id: Allocating OSD id...
command: Running command: /usr/bin/ceph-authtool --gen-print-key
__init__: stderr 
command_with_stdin: Running command with stdin: ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.keyring -i - osd new 9fdce86f-5b14-453d-b78c-cee90efca8b4
command_with_stdin: 0

command_check_call: Running command: /usr/bin/ceph-authtool /var/lib/ceph/osd/ceph-0/keyring --create-keyring --name osd.0 --add-key AQB2Sx9bcxKaKBAAEZI9ytSfb0Yx3ajghDEtEA==
creating /var/lib/ceph/osd/ceph-0/keyring
added entity osd.0 auth auth(auid = 18446744073709551615 key=AQB2Sx9bcxKaKBAAEZI9ytSfb0Yx3ajghDEtEA== with 0 caps)
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/keyring
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/keyring
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/whoami
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/whoami
activate: OSD id is 0
activate: Initializing OSD...
command_check_call: Running command: /usr/bin/ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.keyring mon getmap -o /var/lib/ceph/osd/ceph-0/activate.monmap
got monmap epoch 1
command_check_call: Running command: /usr/bin/ceph-osd --no-mon-config --cluster ceph --mkfs -i 0 --monmap /var/lib/ceph/osd/ceph-0/activate.monmap --osd-data /var/lib/ceph/osd/ceph-0 --osd-uuid 9fdce86f-5b14-453d-b78c-cee90efca8b4 --setuser ceph --setgroup ceph
2018-06-12 04:26:31.890 7f9fd47a91c0 -1 bluestore(/var/lib/ceph/osd/ceph-0/block) _read_bdev_label failed to open /var/lib/ceph/osd/ceph-0/block: (2) No such file or directory
2018-06-12 04:26:31.890 7f9fd47a91c0 -1 bluestore(/var/lib/ceph/osd/ceph-0/block) _read_bdev_label failed to open /var/lib/ceph/osd/ceph-0/block: (2) No such file or directory
2018-06-12 04:26:31.894 7f9fd47a91c0 -1 bdev(0x559203f20700 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:31.894 7f9fd47a91c0 -1 bdev(0x559203f20a80 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:32.430 7f9fd47a91c0 -1 bdev(0x559203f20700 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:32.430 7f9fd47a91c0 -1 bdev(0x559203f21c00 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:32.966 7f9fd47a91c0 -1 bdev(0x559203f21c00 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:32.966 7f9fd47a91c0 -1 bdev(0x559203f20e00 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
command: Running command: /usr/sbin/restorecon -R /var/lib/ceph/osd/ceph-0/active
command: Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/active
activate: ceph osd.0 data dir is ready at /var/lib/ceph/osd/ceph-0
/usr/lib/python2.7/site-packages/ceph_disk/main.py:5721: UserWarning: 
*******************************************************************************
This tool is now deprecated in favor of ceph-volume.
It is recommended to use ceph-volume for OSD deployments. For details see:

    http://docs.ceph.com/docs/master/ceph-volume/#migrating

*******************************************************************************

  warnings.warn(DEPRECATION_WARNING)
ownership of '/var/lib/ceph/osd/ceph-0/ready' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/ceph_fsid' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/whoami' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/bluefs' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/block' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/mkfs_done' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/active' retained as ceph:ceph
changed ownership of '/var/lib/ceph/osd/ceph-0/activate.monmap' from root:root to ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/keyring' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/kv_backend' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/magic' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/fsid' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0/type' retained as ceph:ceph
ownership of '/var/lib/ceph/osd/ceph-0' retained as ceph:ceph
starting osd.0 at - osd_data /var/lib/ceph/osd/ceph-0 /var/lib/ceph/osd/ceph-0/journal
2018-06-12 04:26:33.702 7f05c95a51c0 -1 bdev(0x557015a9a700 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:33.986 7f05c95a51c0 -1 bdev(0x557015a9a700 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:33.986 7f05c95a51c0 -1 bdev(0x557015a9aa80 /var/lib/ceph/osd/ceph-0/block) unable to get device name for /var/lib/ceph/osd/ceph-0/block: (22) Invalid argument
2018-06-12 04:26:34.018 7f05c95a51c0 -1 osd.0 0 log_to_monitors {default=true}
pool 'cephfs_data' created
pool 'cephfs_metadata' created
new fs with metadata pool 2 and data pool 1
changed ownership of '/var/lib/ceph/mds/ceph-demo/keyring' from root:root to ceph:ceph
changed ownership of '/var/lib/ceph/mds/ceph-demo' from root:root to ceph:ceph
starting mds.demo at -
changed ownership of '/var/lib/ceph/radosgw/ceph-rgw.cephst1/keyring' from root:root to ceph:ceph
ownership of '/var/lib/ceph/radosgw/ceph-rgw.cephst1' retained as ceph:ceph
2018-06-12 04:26:38  /entrypoint.sh: Setting up a demo user...
{
    "user_id": "cephst1_uid",
    "display_name": "Ceph demo user",
    "email": "",
    "suspended": 0,
    "max_buckets": 1000,
    "auid": 0,
    "subusers": [],
    "keys": [
        {
            "user": "cephst1_uid",
            "access_key": "cephst1_access_key",
            "secret_key": "cephst1_secret_key"
        }
    ],
    "swift_keys": [],
    "caps": [],
    "op_mask": "read, write, delete",
    "default_placement": "",
    "placement_tags": [],
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "user_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "temp_url_keys": [],
    "type": "rgw",
    "mfa_ids": []
}

{
    "user_id": "cephst1_uid",
    "display_name": "Ceph demo user",
    "email": "",
    "suspended": 0,
    "max_buckets": 1000,
    "auid": 0,
    "subusers": [],
    "keys": [
        {
            "user": "cephst1_uid",
            "access_key": "cephst1_access_key",
            "secret_key": "cephst1_secret_key"
        }
    ],
    "swift_keys": [],
    "caps": [
        {
            "type": "buckets",
            "perm": "*"
        },
        {
            "type": "metadata",
            "perm": "*"
        },
        {
            "type": "usage",
            "perm": "*"
        },
        {
            "type": "users",
            "perm": "*"
        }
    ],
    "op_mask": "read, write, delete",
    "default_placement": "",
    "placement_tags": [],
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "user_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "temp_url_keys": [],
    "type": "rgw",
    "mfa_ids": []
}

2018-06-12 04:26:47  /entrypoint.sh: Creating bucket...
Bucket 's3://cephst1_demo_bucket/' created
curl: (3) <url> malformed     <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
ubuntu@cephst1:~/ceph_demo$

How to reproduce it (minimal and precise):

Pulled the latest ceph/daemon container and ran with

ubuntu@cephst1:~/ceph_demo$ docker pull ceph/daemon:latest

docker run -d \
	--name demo \
	-e MON_IP=172.22.20.175 \
	-e CEPH_PUBLIC_NETWORK=172.22.20.0/24 \
	--net=host \
	-v /var/lib/ceph:/var/lib/ceph \
	-v /etc/ceph:/etc/ceph \
	-e CEPH_DEMO_UID=cephst1_uid \
	-e CEPH_DEMO_ACCESS_KEY=cephst1_access_key \
	-e CEPH_DEMO_SECRET_KEY=cephst1_secret_key \
	-e CEPH_DEMO_BUCKET=cephst1_demo_bucket \
	ceph/daemon \
	demo

ubuntu@cephst1:~/ceph_demo$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
ceph/daemon         latest              71afe972d5d4        16 hours ago        649 MB

**Environment**:
* OS (e.g. from /etc/os-release):
ubuntu@cephst1:~/ceph_demo$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.2 LTS"
ubuntu@cephst1:~/ceph_demo$

* Kernel (e.g. `uname -a`):
ubuntu@cephst1:~/ceph_demo$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.2 LTS"
ubuntu@cephst1:~/ceph_demo$

* Docker version (e.g. `docker version`):
ubuntu@cephst1:~/ceph_demo$ docker version
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64
 Experimental: false
ubuntu@cephst1:~/ceph_demo$ 

* Ceph version (e.g. `ceph -v`):
ubuntu@cephst1:~/ceph_demo$ ceph -v
ceph version 10.2.9 (2ee413f77150c0f375ff6f10edd6c8f9c7d060d0)
ubuntu@cephst1:~/ceph_demo$

**Comment by gfahy** [comment]
Don't know how I can add debug to find out WHAT the malformed culr command is ?

**Comment by leseb** [comment]
We see this in our CI sometimes too, this command is failing: https://github.com/ceph/ceph-container/blob/master/src/daemon/demo.sh#L261

**Comment by rkettelerij** [comment]
I've ran into the same issue. Like @leseb says it's due to curl on the `sree` package which is hosted on github. Luckily this issue is resolved in newer versions of this container, see https://github.com/ceph/ceph-container/commit/3b620574b20184169915408a5ab89e37f2612565#diff-0f625844271d1483107f1b0777882879

Now the `sree` package is embedded in the container image. Which is much better IMHO since having a live container pulling stuff from github at runtime isn't a great practice.

To summarise 3b62057 fixes this issue.
